javascripts/discourse/components/discourse-tag-banners.js

```diff
@@ -0,0 +1,87 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+
+export default class DiscourseTagBanners extends Component {
+  @service store;
+  @service router;
+  @service site;
+  @tracked tag = null;
+  @tracked keepDuringLoadingRoute = false;
+  @tracked isIntersection = false;
+
+  get isVisible() {
+    const route = this.router?.currentRoute;
+
+    if (
+      route &&
+      route.params &&
+      route.params.tag_id &&
+      route.params.tag_id !== "none"
+    ) {
+      this.keepDuringLoadingRoute = true;
+      this.isLoading = false;
+      return true;
+    } else {
+      if (route && route.name.includes("loading")) {
+        return this.keepDuringLoadingRoute;
+      } else {
+        this.keepDuringLoadingRoute = false;
+        this.tag = null;
+        return false;
+      }
+    }
+  }
+
+  formatTagName(tagName = "") {
```

Comment:
I would suggest you make it a private function using `#`:

```javascript
#formatTagName(tagName  = "") {}
```

Which would be then called `this.#formatTagName()`